webapp/src/main/java/com/box/l10n/mojito/service/scheduledjob/jobs/ScheduledThirdPartySync.java

```diff
@@ -0,0 +1,146 @@
+package com.box.l10n.mojito.service.scheduledjob.jobs;
+
+import com.box.l10n.mojito.entity.ScheduledJob;
+import com.box.l10n.mojito.pagerduty.PagerDutyException;
+import com.box.l10n.mojito.pagerduty.PagerDutyIntegrationService;
+import com.box.l10n.mojito.pagerduty.PagerDutyPayload;
+import com.box.l10n.mojito.quartz.QuartzPollableTaskScheduler;
+import com.box.l10n.mojito.service.pollableTask.PollableFuture;
+import com.box.l10n.mojito.service.scheduledjob.IScheduledJob;
+import com.box.l10n.mojito.service.scheduledjob.ScheduledJobRepository;
+import com.box.l10n.mojito.service.thirdparty.ThirdPartySyncJob;
+import com.box.l10n.mojito.service.thirdparty.ThirdPartySyncJobInput;
+import com.box.l10n.mojito.utils.ServerConfig;
+import com.google.common.collect.ImmutableMap;
+import org.apache.commons.lang.text.StrSubstitutor;
+import org.quartz.DisallowConcurrentExecution;
+import org.quartz.JobExecutionContext;
+import org.quartz.JobExecutionException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import org.springframework.web.util.UriComponentsBuilder;
+
+/**
+ * Scheduled job that performs the third party sync for a repository off a cron schedule.
+ *
+ * @author mattwilshire
+ */
+@Component
+@DisallowConcurrentExecution
+public class ScheduledThirdPartySync implements IScheduledJob {
+
+  static Logger logger = LoggerFactory.getLogger(ScheduledThirdPartySync.class);
+
+  @Autowired private QuartzPollableTaskScheduler quartzPollableTaskScheduler;
+  @Autowired private ScheduledJobRepository scheduledJobRepository;
+  @Autowired private PagerDutyIntegrationService pagerDutyIntegrationService;
+
+  @Autowired ServerConfig serverConfig;
+
+  @Value(
+      "${l10n.scheduledJobs.thirdPartySync.notifications.title:MOJITO | Third party sync failed for {repository}}")
```

Comment:
Sorry, I am a bit of a loss here. I know that the `${}` syntax loops in the application properties to get the value but how does the `{repository}` bit work? That's not received from app properties and there doesn't appear to be a variable in scope that it can read from...

Reply:
If you use ${} in the application properties it will evaluate it on startup, if it doesn't find the variable, the process will crash straight away (IIRC). Other approach is to use {0}, {1} but that relies on the message having those parameters which isn't flexible

See:
`String title =
                  StrSubstitutor.replace(
                      notificationTitle,
                      ImmutableMap.of("repository", scheduledJob.getRepository().getName()),
                      "{",
                      "}");`
If {repository} is not in the custom title, that is completely fine it won't do this replace.